app/models/solidus_stripe/payment_method.rb

```diff
@@ -164,11 +164,19 @@ def payment_profiles_supported?
     end
 
     def stripe_dashboard_url(payment)
-      # TODO: handle when the payment doesn't exist yet in Stripe, but we only have the setup intent
-      intent_id = payment.transaction_id
       path_prefix = '/test' if preferred_test_mode
 
-      "https://dashboard.stripe.com#{path_prefix}/payments/#{intent_id}"
+      intent_id =
+        payment.transaction_id ||
+        SolidusStripe::PaymentIntent.where(order: payment.order).pick(:stripe_payment_intent_id) ||
+        SolidusStripe::SetupIntent.where(order: payment.order).pick(:stripe_setup_intent_id)
```

Comment:
Thoughts on extracting the logic to the classes? Like:

```ruby
class SolidusStripe::PaymentIntent
  def self.stripe_id(order)
    where(order: order).pick(:stripe_payment_intent_id)
  end
```

That could help duck-typing at a later stage if we're not ready yet.